cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/generator/GeneratorWrapperFactory.java

```diff
@@ -61,9 +62,12 @@ public class GeneratorWrapperFactory {
    * @throws GeneratorProjectNotExistentException if the generator configuration project does not exist
    * @throws InvalidInputException if the selection includes non supported input types or is composed in a non supported
    *         combination of inputs.
+   * @throws UpgradeTemplatesNotificationException
+   * @throws InvalidConfigurationException
    */
   public static CobiGenWrapper createGenerator(ISelection selection, IProgressMonitor monitor)
-      throws GeneratorCreationException, GeneratorProjectNotExistentException, InvalidInputException {
+      throws GeneratorCreationException, GeneratorProjectNotExistentException, InvalidInputException,
+      InvalidConfigurationException, UpgradeTemplatesNotificationException {
```

Comment:
1. Please rename the exception. What is the exception about? It's not an exception because of "upgrade template notification". That's actually already encoding how you want to handle it. But the exception name should be the cause and not imply any handling as this can differ from case to case.
2. Please make UpgradeTemplatesNotificationException as unchecked exception and remove the throws from all method declaration. As you might have noticed, it's hell of work to continuously declare all exceptions in every method. So normally we should not declare the exception at every method, but handle them at the points of interest. Unchecked exceptions should be preferred.